src/matlab2tikz.m

```diff
@@ -6418,7 +6458,9 @@ function printAll(m2t, env, fid)
     nOpts = size(opts,1);
     c = cell(1,nOpts);
     for k = 1:nOpts
-        if isempty(opts{k,2})
+        if isempty(opts{k,1})
+            c{k} = sprintf('%s', opts{k,2});
```

Comment:
Yes, this was a bit of a hack on my part. To allow the use of ```setColor()``` for marker colors in the ```mark options={red}``` format, I add an entry like this to the options array. I figured it would help me and was unlikely to break anything else. I'd appreciate any alternative suggestions.